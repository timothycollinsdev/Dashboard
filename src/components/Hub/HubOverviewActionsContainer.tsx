import React from "react";
import { useHistory } from "react-router-dom";
import HubOverviewAction from "./HubOverviewAction";

type HubOverviewActionDataType = {
  title: string;
  description: string;
  buttonLabel: string;
  buttonAction: () => void;
};

const HubOverviewActionsContainer = () => {
  const history = useHistory();

  const hubOverviewActionsData: HubOverviewActionDataType[] = [
    {
      title: "Explore",
      description:
        "Find out what you can do in Jina hub and how you platform works.",
      buttonLabel: "Read more",
      buttonAction: () => history.push("/hub/explore"),
    },
    {
      title: "Create",
      // design is not final. Description will probably change.
      description: "Create flow",
      buttonLabel: "Let's Go",
      buttonAction: () => history.push("/flow"),
    },
  ];

  return (
    <>
      {hubOverviewActionsData.map((overviewAction) => (
        <HubOverviewAction
          overviewAction={overviewAction}
          key={overviewAction.title}
        />
      ))}
    </>
  );
};

export default HubOverviewActionsContainer;
