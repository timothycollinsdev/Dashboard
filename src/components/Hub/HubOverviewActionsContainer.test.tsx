import React from "react";
import renderer from "react-test-renderer";
import HubOverviewActionsContainer from "./HubOverviewActionsContainer";

test("actions on hub homepage", () => {
  const component = renderer.create(<HubOverviewActionsContainer />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
