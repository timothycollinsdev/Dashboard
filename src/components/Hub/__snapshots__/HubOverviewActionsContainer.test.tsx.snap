// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions on hub homepage 1`] = `
Array [
  <div
    className="css-1lmpp88"
  >
    <span
      className="css-kysenj"
      data-name="hubOverviewActionTitle"
    >
      Explore
    </span>
    <p>
      Find out what you can do in Jina hub and how you platform works.
    </p>
    <button
      className="css-u2l7it"
      data-name="hubOverviewActionButtonLabel"
      onClick={[Function]}
    >
      Read more
    </button>
  </div>,
  <div
    className="css-1lmpp88"
  >
    <span
      className="css-kysenj"
      data-name="hubOverviewActionTitle"
    >
      Create
    </span>
    <p>
      Create flow
    </p>
    <button
      className="css-u2l7it"
      data-name="hubOverviewActionButtonLabel"
      onClick={[Function]}
    >
      Let's Go
    </button>
  </div>,
]
`;
